apps/io-wallet-user-func/src/android-key-attestation.ts

```diff
@@ -0,0 +1,142 @@
+import * as appInsights from "applicationinsights";
+import { X509Certificate } from "crypto";
+import * as E from "fp-ts/lib/Either";
+import * as O from "fp-ts/lib/Option";
+import * as RTE from "fp-ts/lib/ReaderTaskEither";
+import * as RA from "fp-ts/lib/ReadonlyArray";
+import * as TE from "fp-ts/lib/TaskEither";
+import { pipe } from "fp-ts/lib/function";
+import { WalletInstance } from "io-wallet-common/wallet-instance";
+
+import { AttestationServiceConfiguration } from "./app/config";
+import { validateRevocation } from "./infra/attestation-service/android/attestation";
+import {
+  WalletInstanceRepository,
+  getAllValidWalletInstances,
+  revokeUserWalletInstances,
+} from "./wallet-instance";
+
+const validateKeyAttestationChain = async (
+  walletInstance: WalletInstance,
+  attestationServiceConfiguration: AttestationServiceConfiguration,
+) => {
+  // Check if device details exist and the platform is Android with a valid x509 chain. For iOS is not necessary.
+  if (
+    walletInstance.deviceDetails &&
+    walletInstance.deviceDetails.platform === "android" &&
+    walletInstance.deviceDetails.x509Chain
+  ) {
```

Comment:
I'd either give a name to this condition (ie. `const isAndroidCert = `) or use a runtime type to do this check

Reply:
done